hal/src/rtl872x/interrupts_hal.cpp

```diff
@@ -48,12 +53,87 @@ typedef enum interrupt_state_t {
     INT_STATE_SUSPENDED
 } interrupt_state_t;
 
+#if MODULE_FUNCTION != MOD_FUNC_BOOTLOADER
+
+class InterruptConfig {
+public:
+    struct InterruptCallback {
+        bool operator==(const InterruptCallback& callback) {
+            return handler == callback.handler;
+        }
+
+        hal_interrupt_handler_t handler;
+        void* data;
+        uint32_t priority; // The lower the value, the higher the priority
+    };
+
+    InterruptConfig()
+            : state(INT_STATE_DISABLED),
+              callbacks_() {
+    }
+    ~InterruptConfig() = default;
+
+    int appendHandler(hal_interrupt_handler_t handler, void* data, uint32_t priority) {
+        uint32_t idx = 0;
```

Comment:
Nit: for things like these and needing unsigned type prefer `unsigned` or `size_t`